### DISCUSSION TITLE
While Restoring a Backup having PV into an existing namespace, which has already running pod and pvc bound. In such scenario velero is waiting for restic restores to complete and it's stuck there for long.

### DISCUSSION THREAD

**Discussion by nwakalka** [vmware-tanzu/velero#3258] [Community support Q&A] [answered]
**What steps did you take and what happened:**
[A clear and concise description of what the bug is, and what commands you ran.)

1. Trying to restore a backup into a namespace which has already running pod and pvc bound , in such scenario velero is stucked at **Waiting for all restic restores to complete**

2.  Logs are not rolling and next restore is in state pending. ( waiting for current restore to get complete).

3. Using below command to ran the restore
`nwakalka@nwakalka-virtual-machine:~$ oc rsh -n mcs-customer-backup mcs-velero-7945b48b85-skpcn
Defaulting container name to mcs-velero.
Use 'oc describe pod/mcs-velero-7945b48b85-skpcn -n mcs-customer-backup' to see all of the containers in this pod.`
`sh-4.4$ /velero restore create --from-backup customerbackup-e2e-vwd-iychefsur --include-resources  pods,secrets,persistentvolumeclaims,persistentvolumes,serviceaccounts,configmaps --namespace-mappings e2e-br-src-singlepv-vwd-iychefsur:e2e-br-dest-singlepv-vwd-iychefsur 
Restore request "customerbackup-e2e-vwd-iychefsur-20200930072500" submitted successfully.
Run velero restore describe customerbackup-e2e-vwd-iychefsur-20200930072500 or velero restore logs customerbackup-e2e-vwd-iychefsur-20200930072500 for more details.`
`sh-4.4$ /velero restore logs customerbackup-e2e-vwd-iychefsur-20200930072500
Logs for restore "customerbackup-e2e-vwd-iychefsur-20200930072500" are not available until it's finished processing. Please wait until the restore has a phase of Completed or Failed and try again.
sh-4.4$ 
`

below are logs of velero which is waiting for restic restores to get completed.
`time="2020-09-30T07:25:02Z" level=info msg="Executing item action for pods" logSource="pkg/restore/restore.go:964" restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="[common-restore] Entering common restore plugin" cmd=/plugins/velero-plugins logSource="/go/src/github.com/fusor/openshift-velero-plugin/velero-plugins/common/restore.go:22" pluginName=velero-plugins restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="[common-restore] common restore plugin for e2etest-app-6b744d8774-5md9k" cmd=/plugins/velero-plugins logSource="/go/src/github.com/fusor/openshift-velero-plugin/velero-plugins/common/restore.go:29" pluginName=velero-plugins restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="[GetRegistryInfo] value from imagestream" cmd=/plugins/velero-plugins logSource="/go/src/github.com/fusor/openshift-velero-plugin/velero-plugins/common/shared.go:29" pluginName=velero-plugins restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Executing item action for pods" logSource="pkg/restore/restore.go:964" restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="[pod-restore] Entering Pod restore plugin" cmd=/plugins/velero-plugins logSource="/go/src/github.com/fusor/openshift-velero-plugin/velero-plugins/pod/restore.go:28" pluginName=velero-plugins restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="[pod-restore] pod: e2etest-app-6b744d8774-5md9k" cmd=/plugins/velero-plugins logSource="/go/src/github.com/fusor/openshift-velero-plugin/velero-plugins/pod/restore.go:33" pluginName=velero-plugins restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="[util] container image ref mtr.external.otc.telekomcloud.com/demo/e2etest-app:dev" cmd=/plugins/velero-plugins logSource="/go/src/github.com/fusor/openshift-velero-plugin/velero-plugins/common/util.go:92" pluginName=velero-plugins restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="[util] container image ref mtr.external.otc.telekomcloud.com/paas/mcs-velero-restore-helper:v1.2.0" cmd=/plugins/velero-plugins logSource="/go/src/github.com/fusor/openshift-velero-plugin/velero-plugins/common/util.go:92" pluginName=velero-plugins restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="[pod-restore] skipping restore of pod e2etest-app-6b744d8774-5md9k, has owner references and no restic backup" cmd=/plugins/velero-plugins logSource="/go/src/github.com/fusor/openshift-velero-plugin/velero-plugins/pod/restore.go:49" pluginName=velero-plugins restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Attempting to restore Pod: e2etest-app-6b744d8774-5md9k" logSource="pkg/restore/restore.go:1070" restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Skipping restore of resource because the restore spec excludes it" logSource="pkg/restore/restore.go:398" resource=replicasets.apps restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Skipping restore of resource because the restore spec excludes it" logSource="pkg/restore/restore.go:398" resource=customerbackups.mcs.t-systems.com restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Skipping restore of resource because the restore spec excludes it" logSource="pkg/restore/restore.go:398" resource=customresourcedefinitions.apiextensions.k8s.io restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Skipping restore of resource because the restore spec excludes it" logSource="pkg/restore/restore.go:398" resource=deployments.apps restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Skipping restore of resource because the restore spec excludes it" logSource="pkg/restore/restore.go:398" resource=events restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Skipping restore of resource because the restore spec excludes it" logSource="pkg/restore/restore.go:398" resource=namespaces restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Skipping restore of resource because the restore spec excludes it" logSource="pkg/restore/restore.go:398" resource=replicasets.apps restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Skipping restore of resource because the restore spec excludes it" logSource="pkg/restore/restore.go:398" resource=rolebindings.authorization.openshift.io restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Skipping restore of resource because the restore spec excludes it" logSource="pkg/restore/restore.go:398" resource=rolebindings.rbac.authorization.k8s.io restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
time="2020-09-30T07:25:02Z" level=info msg="Waiting for all restic restores to complete" logSource="pkg/restore/restore.go:470" restore=mcs-customer-backup/customerbackup-e2e-vwd-iychefsur-20200930072500
`

**What did you expect to happen:**

- Expecting to get restore complete or need reason why it's stucked and get timeout.

- Next restore should get completed

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

- When we are using empty/new-namespace, it's working fine.

**Environment:**

- Velero version 1.4.2:

- $ oc version
Client Version: openshift-clients-4.2.0-201910041700
Server Version: 4.4.8
Kubernetes Version: v1.17.1+3f6f40d

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by nwakalka** [comment]
@skriss @ashish-amarnath can someone help on this?

**Comment by ashish-amarnath** [comment]
@nwakalka Apologies for the delayed response. I am going to take a look at this tomorrow.

**Comment by nrb** [comment, marked as answer]
@nwakalka Carlisia is going to take a more in-depth look here, but I think this will not work with how Velero is currently designed.

This sounds like a data-only restore, and for Velero's restic support to work, the pod, PVC, and PV need to be restored together currently.
